### PR TITLE
A bunch of README updates

### LINKED ISSUE #35: Adding footnotes to existing RichTextField features
The README implies to add `features=["footnotes"]` but doesn't give guidance on how to simply extend the list of already existing features.

Do I need to register a hook? Or do I write out the entire features list like:

```
RichTextField(features=["h1", "h2", "h3", "h4", "h5", "h6", "bold", "italic", "ol", "ul", "hr", "link", "document-link", "image", "embed", "footnotes"])
```

### PR DESCRIPTION
**Pull request by benjaoming**
Fixes #35 

[Rendered version](https://github.com/torchbox/wagtail-footnotes/blob/6b0077066a94561609c29d77fcad28e56153d202/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,51 +2,113 @@
...
+### Showing Footnotes in page templates
+
+Update your page templates to include `{% include "wagtail_footnotes/includes/footnotes.html" %}`. You can copy from this template and instead include your own customized version.
+
+### Using footnotes in RichTextField
+
+Update any `RichTextField`s that you want to add footnotes feature. Add `"footnotes"` to the `features` arg for each `RichTextField` that you want to have this functionality. For instance:
+
+```
+class MyPage(Page):
```
Comment (nickmoreton) [inline README.md:50]: Hi @benjaoming Did you test out this usage here?

It doesn't work for me, I also tried it on earlier versions. I don't think it's a use case thats catered for in the package.

RichText field does work to a point (in the admin) but the rendered template doesn't receive any footnotes in the page context.

**Review by nickmoreton — COMMENTED** [review]
I like the restyled docs and was thinking I'd get it merged into the next release.

See the comment on the code example. It's not possible to use this at the moment.

File: README.md (hunk 1 above)
Comment (benjaoming) [reply README.md:50]: I think it's what follows from https://github.com/torchbox/wagtail-footnotes/issues/35 - I don't remember this _not_ working.

I did this in my implementation:

```

{% richtext_footnotes page.get_body %}

{% include "wagtail_footnotes/includes/footnotes.html" %}
```

My `richtext_footnotes` template tag looks like this:

```
@register.simple_tag(takes_context=True)
def richtext_footnotes(context, html):
    """
    example: {% richtext_footnotes page.body|richtext %}

    html: already processed richtext field html
    Assumes "page" in context.
    """
    FIND_FOOTNOTE_TAG = re.compile(r'<footnote id="(.*?)">.*?</footnote>')

    if not isinstance(context.get("page"), Page):
        return html

    page = context["page"]
    if not hasattr(page, "footnotes_list"):
        page.footnotes_list = []
    footnotes = {str(footnote.uuid): footnote for footnote in page.footnotes.all()}

    def replace_tag(match):
        try:
            index = process_footnote(match.group(1), page)
        except (KeyError, ValidationError):
            return ""
        else:
            return f'<a href="#footnote-{index}" id="footnote-source-{index}"><sup>[{index}]</sup></a>'

    def process_footnote(footnote_id, page):
        footnote = footnotes[footnote_id]
        if footnote not in page.footnotes_list:
            page.footnotes_list.append(footnote)
        # Add 1 to the index as footnotes are indexed starting at 1 not 0.
        return page.footnotes_list.index(footnote) + 1

    return mark_safe(FIND_FOOTNOTE_TAG.sub(replace_tag, html))
```

Comment (benjaoming) [reply README.md:50]: Btw, you're right that it doesn't work for the "frontend" - I think I realized this after editing the README. But you're welcome to adapt the above.

I think that I wanted to have some kind of template filter like the above included in the extension. I just dumped some code above, but you could also call the template filter like this:

```
{{ page.body_richtext|richtext|richtext_footnotes %}
```

Comment (nickmoreton) [reply README.md:50]: Yes thats right, the admin works but not the frontend.

I'd agree it seems like a worth while enhancement if you fancy doing it 😄

**Comment by zerolab** [comment]
It only took forever, but this is now merged in afd950eafd7a1e1bd8e235cfb1916180a9bb0022.

Thank you @benjaoming

**Comment by benjaoming** [comment]
Thanks for following up and making it happen @zerolab :partying_face:

**Pull request closed without merging**